open-machine-learning-jupyter-book/ml-fundamentals/classification/more-classifiers.md

```diff
@@ -14,3 +14,221 @@ kernelspec:
+
+```{code-cell}
+cuisines_label_df = cuisines_df['cuisine']
+cuisines_label_df.head()
+```
+
+4\. Drop that `Unnamed: 0` column and the `cuisine` column, calling `drop()`. Save the rest of the data as trainable features:
+
+```{code-cell}
+:tags: [output_scroll]
+
+cuisines_feature_df = cuisines_df.drop(['Unnamed: 0', 'cuisine'], axis=1)
+cuisines_feature_df.head()
+```
+
+Now you are ready to train your model!
+
+## Choosing your classifier
+
+Now that your data is clean and ready for training, you have to decide which algorithm to use for the job. 
+
+Scikit-learn groups classification under Supervised Learning, and in that category you will find many ways to classify. [The variety](https://scikit-learn.org/stable/supervised_learning.html) is quite bewildering at first sight. The following methods all include classification techniques:
+
+- Linear Models
+- Support Vector Machines
+- Stochastic Gradient Descent
+- Nearest Neighbors
+- Gaussian Processes
+- Decision Trees
+- Ensemble methods (voting Classifier)
+- Multiclass and multioutput algorithms (multiclass and multilabel classification, multiclass-multioutput classification)
+
+```{seealso}
+You can also use [neural networks to classify data](https://scikit-learn.org/stable/modules/neural_networks_supervised.html#classification), but that is outside the scope of this section.
+```
+
+### What classifier to go with?
+
+So, which classifier should you choose? Often, running through several and looking for a good result is a way to test. Scikit-learn offers a [side-by-side comparison](https://scikit-learn.org/stable/auto_examples/classification/plot_classifier_comparison.html) on a created dataset, comparing KNeighbors, SVC two ways, GaussianProcessClassifier, DecisionTreeClassifier, RandomForestClassifier, MLPClassifier, AdaBoostClassifier, GaussianNB and QuadraticDiscrinationAnalysis, showing the results visualized:
+
+```{figure} ../../../images/ml-fundamentals/ml-classification/comparison.png
+---
+name: 'comparison of classifiers'
+width: 90%
+---
+Image by [Jen Looper](https://twitter.com/jenlooper)
+```
+
+```{seealso}
+Plots generated on Scikit-learn's documentation.
+
+AutoML solves this problem neatly by running these comparisons in the cloud, allowing you to choose the best algorithm for your data. Try it [here](https://docs.microsoft.com/learn/modules/automate-model-selection-with-azure-automl/?WT.mc_id=academic-77952-leestott).
+```
+
+### A better approach
+
+A better way than wildly guessing, however, is to follow the ideas on this downloadable [ML Cheat sheet](https://docs.microsoft.com/azure/machine-learning/algorithm-cheat-sheet?WT.mc_id=academic-77952-leestott). Here, we discover that, for our multiclass problem, we have some choices:
+
+```{figure} ../../../images/ml-fundamentals/ml-classification/cheatsheet.png
+---
+name: 'comparison of classifiers'
+width: 90%
+---
+Image by [Jen Looper](https://twitter.com/jenlooper)
```

Comment:
Caption should be more accurate. Not need to mention the auther. Add a source if needed.